Bottom: 791212eccf3dbce475eda9edae566cd58458de2e
Top:    6815f7191adc7c5872ea7865d7af6ed44eb90b5d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-14 19:30:06 +0000

select.c: Set an explicit flag to search using when selecting nets by name

If find.c's "TheFlag" variable is not equal to FOUNDFLAG, our code for
selecting nets by name did not work. This could happen, for example - after
a DRC check, where "TheFlag" appears to be left equal to SELECTEDFLAG.


---

diff --git a/src/select.c b/src/select.c
index 4cdc008..ca6a660 100644
--- a/src/select.c
+++ b/src/select.c
@@ -959,6 +959,7 @@ SelectObjectByName (int Type, char *Pattern, bool Flag)
   END_LOOP;
   if (Type & NET_TYPE)
     {
+      SaveFindFlag (FOUNDFLAG);
       InitConnectionLookup ();
       changed = ResetConnections (true) || changed;
 
@@ -983,6 +984,7 @@ SelectObjectByName (int Type, char *Pattern, bool Flag)
       changed = SelectConnection (Flag) || changed;
       changed = ResetConnections (false) || changed;
       FreeConnectionLookupMemory ();
+      RestoreFindFlag ();
     }
 
 #if defined(HAVE_REGCOMP)
